src/_P088_Ping.ino

```diff
@@ -0,0 +1,258 @@
+#ifdef USES_P088
+//#######################################################################################################
+//#################### Plugin 088 ICMP Ping probing ##############
+//#######################################################################################################
+/*
+   Ping tests for hostnames and ips
+   written by https://github.com/nuclearcat
+   Useful to detect strange wifi failures (when it stays connected but not able to reach any ip)
+   and to test other devices for reachability (this is why SendDataOption is enabled)
+   Maintainer: Denys Fedoryshchenko, denys AT nuclearcat.com
+*/
+
+extern "C"
+{
+#include <lwip/raw.h>
+#include <lwip/icmp.h> // needed for icmp packet definitions
+#include <lwip/inet_chksum.h> // needed for inet_chksum()
+#include <lwip/sys.h> // needed for sys_now()
+#include <lwip/netif.h>
+//#include "ESP8266WiFi.h" // needed for WiFi.hostByName()
+}
+
+#define PLUGIN_088
+#define PLUGIN_ID_088             88
+#define PLUGIN_NAME_088           "Communication - Ping"
+#define PLUGIN_VALUENAME1_088     "Fails"
+#define PLUGIN_088_HOSTNAME_SIZE  64
+#define PLUGIN_088_MAX_INSTANCES  8
+
+/* Most efficient data packing if we order from place elements from largest to smallest */
+struct P088_data_struct {
+  ip_addr_t destIPAddress[TASKS_MAX];
+  uint32_t idseq[TASKS_MAX];
```

Comment:
For `TASKS_MAX` of 12, this already takes about 100 bytes of memory, when at least 1 of these plugins is enabled.
Please have a look at the other plugins using `PluginTaskData_base` to see what I tried to make to support multiple instances of plugins.
You still need a pointer to the callback function, but at least it does not allocate 12x as much data as needed if you're only running 1 instance of the plugin. (and some people are running with changed settings files to support 24 tasks)